unittests/Feature/FeatureModelTransaction.cpp

```diff
@@ -832,4 +832,81 @@ TEST_F(FeatureModelRemoveFeatureTransactionTest,
+protected:
+  void SetUp() override {
+    FeatureModelBuilder B;
+    B.makeFeature<BinaryFeature>("a", true);
+    FM = B.buildFeatureModel();
+    ASSERT_TRUE(FM);
+  }
+
+  std::unique_ptr<FeatureModel> FM;
+};
+
+TEST_F(FeatureModelTransactionTest, addFeatureToModel) {
+  size_t FMSizeBefore = FM->size();
+
+  vara::feature::addFeature(*FM,
+                            std::move(std::make_unique<BinaryFeature>("ab")),
+                            FM->getFeature("a")); // committed automatically
+
+  // Changes should be visible on the new model
+  EXPECT_EQ(FMSizeBefore, FM->size() - 1);
+  EXPECT_TRUE(FM->getFeature("a"));
+  EXPECT_TRUE(llvm::isa<RootFeature>(FM->getFeature("a")->getParentFeature()));
+  EXPECT_TRUE(FM->getFeature("ab")); // Change should be visible
+  EXPECT_EQ(FM->getFeature("a"), FM->getFeature("ab")->getParentFeature());
+}
+
+TEST_F(FeatureModelTransactionTest, addFeaturesToModel) {
+  size_t FMSizeBefore = FM->size();
+
+  std::vector<std::tuple<std::unique_ptr<Feature>, Feature *>> NewFeatures;
+  NewFeatures.emplace_back(std::make_tuple(
+      std::make_unique<BinaryFeature>("ab"), FM->getFeature("a")));
+  NewFeatures.emplace_back(std::make_tuple(
+      std::make_unique<BinaryFeature>("ac"), FM->getFeature("a")));
+
+  vara::feature::addFeatures(*FM,
+                             std::move(NewFeatures)); // committed automatically
+
+  // Changes should be visible on the new model
+  EXPECT_EQ(FMSizeBefore, FM->size() - 2);
+  EXPECT_TRUE(FM->getFeature("a"));
+  EXPECT_TRUE(llvm::isa<RootFeature>(FM->getFeature("a")->getParentFeature()));
+  EXPECT_TRUE(FM->getFeature("ab")); // Change should be visible
+  EXPECT_EQ(FM->getFeature("a"), FM->getFeature("ab")->getParentFeature());
+  EXPECT_TRUE(FM->getFeature("ac")); // Change should be visible
+  EXPECT_EQ(FM->getFeature("a"), FM->getFeature("ac")->getParentFeature());
+}
+
+TEST_F(FeatureModelTransactionTest, removeFeatureFromModel) {
+  size_t FMSizeBefore = FM->size();
+
+  // Prepare Model with two Features
+  std::vector<std::tuple<std::unique_ptr<Feature>, Feature *>> NewFeatures;
+  NewFeatures.emplace_back(std::make_tuple(
+      std::make_unique<BinaryFeature>("ab"), FM->getFeature("a")));
+  NewFeatures.emplace_back(std::make_tuple(
+      std::make_unique<BinaryFeature>("ac"), FM->getFeature("a")));
+  vara::feature::addFeatures(*FM,
+                             std::move(NewFeatures)); // committed automatically
+
+  // Remove one Feature
+  // std::variant<std::string, Feature *>
+  vara::feature::removeFeature(*FM, FM->getFeature("ab"));
+
+  // Changes should be visible on the new model
+  EXPECT_EQ(FMSizeBefore, FM->size() - 1);
+  EXPECT_TRUE(FM->getFeature("a"));
+  EXPECT_TRUE(llvm::isa<RootFeature>(FM->getFeature("a")->getParentFeature()));
+  EXPECT_TRUE(FM->getFeature("ac")); // Change should be visible
+  EXPECT_EQ(FM->getFeature("a"), FM->getFeature("ac")->getParentFeature());
+}
```

Comment:
This test does not test what it wants to test. We want to guarantee that `"ab"` is remove after being added. But we never check for `"ab"`. If we remove the implementation of `removeFeature` this test still passes.